helm/cluster-cloud-director/templates/_helpers.tpl

```diff
@@ -219,3 +220,65 @@ taints:
 {{- $outerScope := . }}
 {{- include "mtRevision" (merge (dict "currentClass" .Values.controlPlane) $outerScope.Values) }}
 {{- end -}}
+
+{{/*
+Join elements of a string array by quoting elements and using comma as seperator.
+*/}}
+{{- define "commaJoinedQuotedList" -}}
+{{- $list := list }}
+{{- range . }}
+{{- $list = append $list (printf "\"%s\"" .) }}
+{{- end }}
+{{- join ", " $list }}
+{{- end }}
+
+{{/*
+Generate a stanza for KubeAdmConfig and KubeAdmControlPlane in order to 
+mount containerd configuration for registry configuration in nodes.
+*/}}
+{{- define "registryFiles" -}}
+{{- if .Values.registry.configure -}}
+- path: /etc/containerd/conf.d/registry-config.toml
+  permissions: "0600"
+  contentFrom:
+    secret:
+      name: {{ include "registrySecretName" $ }}
+      key: registry-config.toml
+{{- end -}}
+{{- end -}}
+
+{{/*
+Generate the content of /etc/containerd/conf.d/registry-config.toml in nodes
+for registry configuration
+*/}}
+{{- define "registrySecretContent" -}}
+version = 2
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+{{- range .Values.registry.mirrors }}
+[plugins."io.containerd.grpc.v1.cri".registry.mirrors."{{.name}}"]
+endpoint = [ {{ include "commaJoinedQuotedList" .endpoints }} ]
```

Comment:
Isn't that simpler?

```suggestion
endpoint = [ {{ if .endpoints }}"{{ join "\", \"" .endpoints }}"{{ end }} ]
```

Reply:
I tried to find this solution and even found this answer https://stackoverflow.com/a/59092458 but it didn't work when I tried. I tried it now again and it works :D I don't know what was wrong on my first try. Escaping was not working. Anyway. I am updating my commit.